src/Bicep.Core/TypeSystem/TypeSymbol.cs

```diff
@@ -18,6 +18,8 @@ protected TypeSymbol(string name) : base(name)
 
         public TypeSymbol Type => this;
 
+        public virtual TypeSymbolValidationFlags ValidationFlags { get; } = TypeSymbolValidationFlags.Default;
```

Comment:
Making this virtual with a default to avoid having to add an extra param to constructors across the codebase. It's a bit weird, as you may want to declare e.g. a primitive type with flags. In practice, this doesn't affect the `TypeValidator` as the errors are being raised for complex types (named objects, typed arrays, discriminated objects).

Reply:
Sounds good.

---
In reply to: [497608527](https://github.com/Azure/bicep/pull/578#discussion_r497608527) [](ancestors = 497608527)